internal/controllers/user_devices_controller.go

```diff
@@ -1580,30 +1607,69 @@ func (udc *UserDevicesController) PostMintDevice(c *fiber.Ctx) error {
 				return err
 			}
 
-			return client.MintVehicleAndSDign(requestID, contracts.MintVehicleAndSdInput{
-				ManufacturerNode: makeTokenID,
-				Owner:            realAddr,
-				AttrInfoPairsVehicle: []contracts.AttributeInfoPair{
-					{Attribute: "Make", Info: deviceMake},
-					{Attribute: "Model", Info: deviceModel},
-					{Attribute: "Year", Info: deviceYear},
-				},
+			if udc.Settings.IsProduction() {
+
+				return client.MintVehicleAndSdSign(requestID, contracts.MintVehicleAndSdInput{
+					ManufacturerNode:    makeTokenID,
+					Owner:               realAddr,
+					IntegrationNode:     new(big.Int).SetUint64(intID),
+					VehicleOwnerSig:     sigBytes,
+					SyntheticDeviceSig:  sign,
+					SyntheticDeviceAddr: common.BytesToAddress(addr),
+					AttrInfoPairsDevice: []contracts.AttributeInfoPair{
+						{
+							Attribute: "Make",
+							Info:      dd.Make.Name,
+						},
+						{
+							Attribute: "Model",
+							Info:      dd.Type.Model,
+						},
+						{
+							Attribute: "Year",
+							Info:      strconv.Itoa(int(dd.Type.Year)),
+						},
+					},
+				})
+			}
+
+			return client.MintVehicleAndSdWithDeviceDefinitionSign(requestID, contracts.MintVehicleAndSdWithDdInput{
+				ManufacturerNode:    makeTokenID,
+				Owner:               realAddr,
+				DeviceDefinitionId:  dd.NameSlug,
 				IntegrationNode:     new(big.Int).SetUint64(intID),
 				VehicleOwnerSig:     sigBytes,
 				SyntheticDeviceSig:  sign,
 				SyntheticDeviceAddr: common.BytesToAddress(addr),
-				AttrInfoPairsDevice: []contracts.AttributeInfoPair{},
+				AttrInfoPairsDevice: []contracts.AttributeInfoPair{
+					{
+						Attribute: "Make",
+						Info:      dd.Make.Name,
+					},
+					{
+						Attribute: "Model",
+						Info:      dd.Type.Model,
+					},
+					{
+						Attribute: "Year",
+						Info:      strconv.Itoa(int(dd.Type.Year)),
+					},
+				},
 			})
 		}
 	}
 
 	logger.Info().Msgf("Submitted metatransaction request %s", requestID)
 
-	return client.MintVehicleSign(requestID, makeTokenID, realAddr, []contracts.AttributeInfoPair{
-		{Attribute: "Make", Info: deviceMake},
-		{Attribute: "Model", Info: deviceModel},
-		{Attribute: "Year", Info: deviceYear},
-	}, sigBytes)
+	if udc.Settings.IsProduction() {
+		return client.MintVehicleSign(requestID, makeTokenID, realAddr, []contracts.AttributeInfoPair{
+			{Attribute: "Make", Info: dd.Make.Name},
+			{Attribute: "Model", Info: dd.Type.Model},
+			{Attribute: "Year", Info: strconv.Itoa(int(dd.Type.Year))},
+		}, sigBytes)
+	}
+
+	return client.MintVehicleWithDeviceDefinitionSign(requestID, makeTokenID, realAddr, dd.NameSlug, sigBytes)
```

Comment:
Isn't there a version of this with attributes now? Do you need to update?

Reply:
https://github.com/DIMO-Network/dimo-identity/blob/811a89081cc89c821cb0b37c2ac6796731af743d/contracts/implementations/nodes/VehicleInternal.sol#L21